Rocket.API/Utils/TaskDispatcher.cs

```diff
@@ -8,9 +8,16 @@ namespace Rocket.API.Utils
     public class TaskDispatcher : MonoBehaviour
     {
         public static TaskDispatcher Instance { get; private set; }
-        private static readonly List<Action> QueuedMainActions = new List<Action>();
-        private static readonly List<Action> QueuedMainFixedActions = new List<Action>();
-        private static readonly List<Action> QueuedAsyncActions = new List<Action>();
+
+        private static readonly Queue<Action> QueuedMainActions = new Queue<Action>();
+        private static readonly Queue<Action> QueuedMainActionsBuffer = new Queue<Action>();
```

Comment:
Why is there a second queue for queued main actions?

Reply:
The second set of queues act as a buffer to allow the thread requesting the action on the other thread to allow execution even if the other thread is working on its current queue.

Reply:
This would mainly help prevent a thread from stalling when it tries to add something to the queue of a different thread that's performing a heavily I/O based function that could take longer than a few milliseconds.

Reply:
Well then don't lock it that long I suppose. Dequeue what's in the queue and leave the lock, then do the action. There is no need for a second collection.

Reply:
Without the second collection, it would have to re-enter the lock multiple times per tick or only execute one action per tick.

Reply:
Why not keep it as list, enter the lock, copy the list to a local variable in the main thread, leave the lock and handle the local list.

Reply:
By using a local list, wouldn't the object have to be re instantiated every tick? At least on the main thread where the method does eventually exit to allow the tread to handle other things. I can change them back to lists also, but a queue seems like it is more designed for a job like this.

Reply:
If you intend to fetch all entries anyway i suppose a queue is not making the job easier. One could also implement a custom collection that returns new lists when the thread is asking for the data & clears its contents automatically, not blocking the list in the time the resulting collection is iterated.